cluster-autoscaler/processors/provreq/injector.go

```diff
@@ -44,6 +46,67 @@ type ProvisioningRequestPodsInjector struct {
 	clock  clock.PassiveClock
 }
 
+// IsSupportedClass checks if the ProvisioningRequest is supported by the processor.
+func (p *ProvisioningRequestPodsInjector) IsSupportedClass(provisioningClassName string, supportedClasses []string) bool {
+	if len(supportedClasses) == 0 {
+	_, found := provisioningrequest.SupportedProvisioningClasses[provisioningClassName]
+	return found
+	}
+
+	for _, supportedClass := range supportedClasses {
```

Comment:
The second parameter is so that a custom condition for filtering provisioning request classes could be implemented. Refactored the logic to a separate helper method.